syntax = "proto3";

package internal.proto;

message InternalSliceRequest {
    uint64 slice_partition = 1;
    uint64 data_start = 2;
    uint64 data_len = 3;
}

message InternalRequest {
  bytes model_name = 1;
  repeated InternalSliceRequest slice_request = 2;
}

message InternalResponse {
  int32 status = 1; // 0，返回结果ok，-1 服务发生内部错误
  repeated bytes slice_data = 2;
}

service InternalModelService {
  rpc Get(InternalRequest) returns (InternalResponse) {}
}